### PR TITLE
Update check_connection method to use empty document list index request

### LINKED ISSUE #14: Fix Workplace Search test failure on 8.x due to endpoint elimination
**Build scan**:
https://app.travis-ci.com/github/logstash-plugins/logstash-integration-elastic_enterprise_search/jobs/551568629

**Repro line**:

**Reproduces locally?**:
yes:
```
export ELASTIC_STACK_VERSION=8.x
export SNAPSHOT=true
.ci/docker-setup.sh
.ci/docker-run.sh
```

**Applicable branches**:
`main`

**Failure history**:

**Failure excerpt**:
```
logstash_1_33c6a5d2aa4f | Failures:
logstash_1_33c6a5d2aa4f | 
logstash_1_33c6a5d2aa4f |   1) indexing against running Workplace Search single event should be indexed
logstash_1_33c6a5d2aa4f |      Failure/Error: raise ::LogStash::ConfigurationError.new("Failed to connect to Workplace Search. Error: 404. Please check if url '#{@url}' is correct and you've created a source with ID '#{@source}'")
logstash_1_33c6a5d2aa4f |      
logstash_1_33c6a5d2aa4f |      LogStash::ConfigurationError:
logstash_1_33c6a5d2aa4f |        Failed to connect to Workplace Search. Error: 404. Please check if url 'http://enterprise_search:3002' is correct and you've created a source with ID '61b21f6aec516863bfbccf99'
logstash_1_33c6a5d2aa4f |      # ./lib/logstash/outputs/elastic_workplace_search.rb:47:in `register'
logstash_1_33c6a5d2aa4f |      # ./spec/integration/outputs/elastic_workplace_search_spec.rb:35:in `block in <main>'
logstash_1_33c6a5d2aa4f |      # ------------------
logstash_1_33c6a5d2aa4f |      # --- Caused by: ---
logstash_1_33c6a5d2aa4f |      # Elastic::WorkplaceSearch::NonExistentRecord:
logstash_1_33c6a5d2aa4f |      #   Elastic::WorkplaceSearch::NonExistentRecord
logstash_1_33c6a5d2aa4f |      #   ./lib/logstash/outputs/elastic_workplace_search.rb:104:in `check_connection!'
logstash_1_33c6a5d2aa4f | 
logstash_1_33c6a5d2aa4f |   2) indexing against running Workplace Search multiple events all should be indexed
logstash_1_33c6a5d2aa4f |      Failure/Error: raise ::LogStash::ConfigurationError.new("Failed to connect to Workplace Search. Error: 404. Please check if url '#{@url}' is correct and you've created a source with ID '#{@source}'")
logstash_1_33c6a5d2aa4f |      
logstash_1_33c6a5d2aa4f |      LogStash::ConfigurationError:
logstash_1_33c6a5d2aa4f |        Failed to connect to Workplace Search. Error: 404. Please check if url 'http://enterprise_search:3002' is correct and you've created a source with ID '61b21f6fec5168d5a2bccf9b'
logstash_1_33c6a5d2aa4f |      # ./lib/logstash/outputs/elastic_workplace_search.rb:47:in `register'
logstash_1_33c6a5d2aa4f |      # ./spec/integration/outputs/elastic_workplace_search_spec.rb:35:in `block in /usr/share/plugins/plugin/spec/integration/outputs/elastic_workplace_search_spec.rb'
logstash_1_33c6a5d2aa4f |      # ------------------
logstash_1_33c6a5d2aa4f |      # --- Caused by: ---
logstash_1_33c6a5d2aa4f |      # Elastic::WorkplaceSearch::NonExistentRecord:
logstash_1_33c6a5d2aa4f |      #   Elastic::WorkplaceSearch::NonExistentRecord
logstash_1_33c6a5d2aa4f |      #   ./lib/logstash/outputs/elastic_workplace_search.rb:104:in `check_connection!'
logstash_1_33c6a5d2aa4f | 
logstash_1_33c6a5d2aa4f | Finished in 28.88 seconds (files took 9.35 seconds to load)
logstash_1_33c6a5d2aa4f | 8 examples, 2 failures
logstash_1_33c6a5d2aa4f | 
logstash_1_33c6a5d2aa4f | Failed examples:
logstash_1_33c6a5d2aa4f | 
logstash_1_33c6a5d2aa4f | rspec ./spec/integration/outputs/elastic_workplace_search_spec.rb:40 # indexing against running Workplace Search single event should be indexed
logstash_1_33c6a5d2aa4f | rspec ./spec/integration/outputs/elastic_workplace_search_spec.rb:57 # indexing against running Workplace Search multiple events all should be indexed
```

**Solution**
The endpoint used by `list_all_permissions` https://github.com/logstash-plugins/logstash-integration-elastic_enterprise_search/blob/6c8fba850849f8b8a14d29e62722a69a04f1ac69/lib/logstash/outputs/elastic_workplace_search.rb#L104 has been removed in `8.x`.

The alternative to check connectivity goodness is to index an empty set of documents with `index_documents`, OR remove a document (`destroy_documents`)with a not-existent id, generated with `SecuyreRandom.uuid`

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

Change implementation of connectivity check method to be compatible with version `8.0` of Workplace.

## What does this PR do?
Change implementation of `check_connection!` method, replacing the call to `list_all_permissions` which doesn't have anymore a peer endpoint in `8.0` to an empty document list `index_documents`.
Fixes also the integration tests for Workplace Search version 8, switching API to retrieve the `access_token`

## Why is it important/What is the impact to the user?

Fixes a compatibility problem with Workplace Search `8.0`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files (and/or docker env variables)~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [x] Run the failure path described in issue #14 

## How to test this PR locally

```sh
export ELASTIC_STACK_VERSION=8.x
export SNAPSHOT=true
.ci/docker-setup.sh
.ci/docker-run.sh
```

## Related issues

- Fixes #14

### CODE REVIEW

File: .ci/Dockerfile [hunk 1]
```diff
@@ -8,6 +8,7 @@ COPY --chown=logstash:logstash *.gemspec VERSION* version* /usr/share/plugins/pl
 RUN cp /usr/share/logstash/logstash-core/versions-gem-copy.yml /usr/share/logstash/versions.yml
 ENV PATH="${PATH}:/usr/share/logstash/vendor/jruby/bin:/usr/share/logstash/jdk/bin"
 ENV LOGSTASH_SOURCE="1"
+ARG ELASTIC_STACK_VERSION
```
Comment (andsel) [inline .ci/Dockerfile:11]: Note for reviewer:
without this the `ENV['ELASTIC_STACK_VERSION']` doesn't contain value

File: spec/integration/outputs/elastic_workplace_search_spec.rb [hunk 2]
```diff
@@ -8,25 +8,69 @@
...
-      "Authorization" => "Basic #{auth}"
-    )
-    JSON.load(response.body)
+          "#{url}/api/ws/v1/sources",
+           JSON.dump("service_type" => "custom", "name" => "whatever"),
+          {"Content-Type" => "application/json",
+          "Accept" => "application/json",
+          "Authorization" => "Bearer #{access_token}"}
+        )
+    puts "DBG>> source_id response body: #{response.body}"
```
Comment (kaisecheng) [inline spec/integration/outputs/elastic_workplace_search_spec.rb:64]: do you really want to print value here?

**Review by kaisecheng — APPROVED** [review]
tested locally. fixed the issue as it tells. LGTM

File: spec/integration/outputs/elastic_workplace_search_spec.rb (hunk 2 above)
Comment (andsel) [reply spec/integration/outputs/elastic_workplace_search_spec.rb:64]: No, my bad I didn't remove it

**Pull request merged**